lib/ansible/executor/task_queue_manager.py

```diff
@@ -317,6 +351,10 @@ def has_dead_workers(self):
         return defunct
 
     def send_callback(self, method_name, *args, **kwargs):
+        with self._callback_lock:
```

Comment:
Why the nested impl instead of just doing the lock inline or a lock decorator?

Reply:
Honestly, I don't like this pattern that I went with, it was a preference of a former core dev over using decorators.  Although I'd prefer a decorator, and I think I can address that.

Reply:
I addressed this, but I may have taken it too far. I created a new generic decorator, and used it here, rather than a use case specific implementation in TQM.